### PR TITLE
ABC --> CAS conversion

### LINKED ISSUE #22: Add ABC --> CAS conversion
See https://github.com/AllenInstitute/abc_atlas_access

https://github.com/dosumis/ABC_2_CAS/blob/main/notebooks/ABC_2_CAS.ipynb. (ignore all the stuff with old annotations & CL terms)

3 tables
-  labelset details
image
- hierarchy of ranked labelsets
 image
- cluster membership
image

### PR DESCRIPTION
**Pull request by ubyndr**
Resolves #22

### CODE REVIEW

File: src/cas/abc_cas_converter.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    matrix_file_id = None
+    cellannotation_schema_version = None
+    cellannotation_timestamp = None
+    cellannotation_version = None
+    cellannotation_url = None
+    author_name = None
+    author_contact = None
+    orcid = None
+    cas = {
+        "matrix_file_id": matrix_file_id,
```
Comment (hkir-dev) [inline src/cas/abc_cas_converter.py:220]: Can we inline above variable declarations as well?

**Review by hkir-dev — CHANGES_REQUESTED** [review]

File: src/cas/abc_cas_converter.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+def init_metadata() -> Dict[str, Any]:
+    """
+    Initializes metadata for Cell Annotation Schema (CAS).
+
+    Returns:
+        Dict[str, Any]: Metadata dictionary containing default values for various fields.
+    """
+    # TODO These needs proper assignments
+    matrix_file_id = None
+    cellannotation_schema_version = None
```
[the remark this answers is not in the document]
Comment (ubyndr) [reply src/cas/abc_cas_converter.py]: >Use of pkg_resources is deprecated in favor of [importlib.resources](https://docs.python.org/3.11/library/importlib.resources.html#module-importlib.resources), [importlib.metadata](https://docs.python.org/3.11/library/importlib.metadata.html#module-importlib.metadata) and their backports ([importlib_resources](https://pypi.org/project/importlib_resources), [importlib_metadata](https://pypi.org/project/importlib_metadata)). Some useful APIs are also provided by [packaging](https://pypi.org/project/packaging) (e.g. requirements and version parsing). Users should refrain from new usage of pkg_resources and should work to port to importlib-based solutions.

Used the following instead
```python
from importlib.metadata import version
version("cell-annotation-schema")
```

File: src/cas/abc_cas_converter.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        synonyms = None
+        category_fullname = None
+        category_cell_ontology_exists = None
+        category_cell_ontology_term_id = None
+        category_cell_ontology_term = None
+        cell_set_accession = row.label
+        parent_cell_set_accession = row.parent_term_label
+
+        anno = {
+            "labelset": labelset,
```
[the remark this answers is not in the document]
Comment (ubyndr) [reply src/cas/abc_cas_converter.py:162]: I would like to keep it as it is for now, as transitioning to a CAS object would necessitate a more in-depth refactoring of our code

File: src/cas/abc_cas_converter.py (hunk 1 above)
Comment (ubyndr) [reply src/cas/abc_cas_converter.py:221]: I would like to keep it as it is for now, as transitioning to a CAS object would necessitate a more in-depth refactoring of our code

**Comment by dosumis** [comment]
Seems to work OK, but noticed it fills in the cell_ontology_term field with the label. Can you fix that?

image

**Review by hkir-dev — APPROVED** [review]

**Pull request merged**